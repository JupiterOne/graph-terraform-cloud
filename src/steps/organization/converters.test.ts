import {
  getMockEntitlementSet,
  getMockOrganization,
  getMockOrganizationWorkspace,
  getMockUser,
} from '../../../test/mocks';
import {
  createOrganizationEntitlementSetEntity,
  createOrganizationEntity,
  createOrganizationMemberEntity,
  createOrganizationWorkspaceEntity,
} from './converters';

describe('#createOrganizationEntity', () => {
  test('should convert to entity', () => {
    expect(createOrganizationEntity(getMockOrganization())).toMatchSnapshot();
  });
});

describe('#createOrganizationMemberEntity', () => {
  test('should convert to entity', () => {
    expect(
      createOrganizationMemberEntity({
        userId: 'test-id',
        userData: getMockUser(),
      }),
    ).toMatchSnapshot();
  });
});

describe('#createOrganizationWorkspaceEntity', () => {
  test('should convert to entity', () => {
    expect(
      createOrganizationWorkspaceEntity(
        'workspace-id',
        getMockOrganizationWorkspace(),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createOrganizationEntitlementSetEntity', () => {
  test('should convert to entity', () => {
    expect(
      createOrganizationEntitlementSetEntity(
        'organization-id',
        getMockEntitlementSet(),
      ),
    ).toMatchSnapshot();
  });
});
