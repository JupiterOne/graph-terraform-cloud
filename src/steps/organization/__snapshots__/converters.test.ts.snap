// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createOrganizationEntitlementSetEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Entity",
  ],
  "_key": "entitlement-set-organization-id",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "agents": false,
        "auditLogging": false,
        "configurationDesigner": true,
        "costEstimation": false,
        "id": "org-yMai1ZUhS39WMAU9",
        "operations": true,
        "privateModuleRegistry": true,
        "selfServeBilling": true,
        "sentinel": false,
        "sso": false,
        "stateStorage": true,
        "teams": false,
        "usageReporting": false,
        "userLimit": 5,
        "vcsIntegrations": true,
      },
    },
  ],
  "_type": "tfe_entitlement_set",
  "agents": false,
  "auditLogging": false,
  "configurationDesigner": true,
  "costEstimation": false,
  "createdOn": undefined,
  "displayName": "entitlement-set-organization-id",
  "id": "org-yMai1ZUhS39WMAU9",
  "name": "entitlement-set-organization-id",
  "operations": true,
  "privateModuleRegistry": true,
  "selfServeBilling": true,
  "sentinel": false,
  "sso": false,
  "stateStorage": true,
  "teams": false,
  "usageReporting": false,
  "userLimit": 5,
  "vcsIntegrations": true,
}
`;

exports[`#createOrganizationEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Organization",
  ],
  "_key": "org-yMai1ZUhS39WMAU9",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "collaboratorAuthPolicy": "password",
        "costEstimationEnabled": false,
        "createdAt": "2021-02-17T21:24:28.929Z",
        "email": "austin.kelleher@jupiterone.com",
        "externalId": "org-yMai1ZUhS39WMAU9",
        "fairRunQueuingEnabled": true,
        "name": "jupiterone",
        "ownersTeamSamlRoleId": null,
        "permissions": Object {
          "canAccessViaTeams": true,
          "canCreateModule": false,
          "canCreateProvider": false,
          "canCreateTeam": false,
          "canCreateWorkspace": true,
          "canDestroy": false,
          "canManageCustomProviders": false,
          "canManagePublicModules": false,
          "canManagePublicProviders": false,
          "canManageSso": false,
          "canManageSubscription": false,
          "canManageTags": true,
          "canManageUsers": true,
          "canManageVarsets": true,
          "canReadVarsets": true,
          "canStartTrial": true,
          "canTraverse": true,
          "canUpdate": false,
          "canUpdateAgentPools": false,
          "canUpdateApiToken": true,
          "canUpdateOauth": true,
          "canUpdateSentinel": false,
          "canUpdateSshKeys": false,
        },
        "planExpired": false,
        "planIdentifier": "free",
        "planIsEnterprise": false,
        "planIsTrial": false,
        "samlEnabled": false,
        "sendPassingStatusesForUntriggeredSpeculativePlans": true,
        "twoFactorConformant": false,
      },
    },
  ],
  "_type": "tfe_organization",
  "collaboratorAuthPolicy": "password",
  "costEstimationEnabled": false,
  "createdOn": 1613597068929,
  "displayName": "jupiterone",
  "email": "austin.kelleher@jupiterone.com",
  "fairRunQueuingEnabled": true,
  "id": "org-yMai1ZUhS39WMAU9",
  "mfaEnabled": false,
  "name": "jupiterone",
  "ownersTeamSamlRoleId": null,
  "permissions.canAccessViaTeams": true,
  "permissions.canCreateModule": false,
  "permissions.canCreateProvider": false,
  "permissions.canCreateTeam": false,
  "permissions.canCreateWorkspace": true,
  "permissions.canDestroy": false,
  "permissions.canManageCustomProviders": false,
  "permissions.canManagePublicModules": false,
  "permissions.canManagePublicProviders": false,
  "permissions.canManageSso": false,
  "permissions.canManageSubscription": false,
  "permissions.canManageTags": true,
  "permissions.canManageUsers": true,
  "permissions.canManageVarsets": true,
  "permissions.canReadVarsets": true,
  "permissions.canStartTrial": true,
  "permissions.canTraverse": true,
  "permissions.canUpdate": false,
  "permissions.canUpdateAgentPools": false,
  "permissions.canUpdateApiToken": true,
  "permissions.canUpdateOauth": true,
  "permissions.canUpdateSentinel": false,
  "permissions.canUpdateSshKeys": false,
  "planExpired": false,
  "planExpiresAt": undefined,
  "planIsEnterprise": false,
  "planIsTrial": false,
  "samlEnabled": false,
  "sessionRemember": undefined,
  "sessionTimeout": undefined,
  "webLink": "https://app.terraform.io/app/jupiterone/workspaces",
}
`;

exports[`#createOrganizationMemberEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "User",
  ],
  "_key": "test-id",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "avatarUrl": "https://www.gravatar.com/avatar/c726702df6456d9affb1eb8d9d37a9de?s=100&d=mm",
        "email": "api-org-test@hashicorp.com",
        "enterpriseSupport": false,
        "hasGitHubAppToken": false,
        "id": "user-TSVGDCJGAZZZZZZ",
        "isConfirmed": true,
        "isServiceAccount": true,
        "isSiteAdmin": false,
        "isSsoLogin": false,
        "isSudo": false,
        "permissions": Object {
          "canChangeEmail": true,
          "canChangeUsername": true,
          "canCreateOrganizations": false,
          "canManageUserTokens": false,
        },
        "twoFactor": Object {
          "enabled": false,
          "verified": false,
        },
        "username": "api-org-test",
      },
    },
  ],
  "_type": "tfe_user",
  "avatarUrl": "https://www.gravatar.com/avatar/c726702df6456d9affb1eb8d9d37a9de?s=100&d=mm",
  "createdOn": undefined,
  "displayName": "api-org-test",
  "email": "api-org-test@hashicorp.com",
  "id": "test-id",
  "isServiceAccount": true,
  "mfaEnabled": false,
  "mfaVerified": false,
  "name": "api-org-test",
  "permissions.canChangeEmail": true,
  "permissions.canChangeUsername": true,
  "permissions.canCreateOrganizations": false,
  "permissions.canManageUserTokens": false,
  "username": "api-org-test",
}
`;

exports[`#createOrganizationWorkspaceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Project",
  ],
  "_key": "workspace-id",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "actions": Object {
          "isDestroyable": true,
        },
        "allowDestroyPlan": true,
        "applyDurationAverage": 34000,
        "autoApply": false,
        "createdAt": "2021-02-17T21:26:58.799Z",
        "environment": "default",
        "executionMode": "remote",
        "fileTriggersEnabled": false,
        "globalRemoteState": true,
        "latestChangeAt": "2021-05-28T17:39:48.509Z",
        "locked": false,
        "name": "integration-development-google-cloud",
        "operations": true,
        "permissions": Object {
          "canCreateStateVersions": false,
          "canDestroy": true,
          "canForceUnlock": false,
          "canLock": false,
          "canManageTags": true,
          "canQueueApply": false,
          "canQueueDestroy": false,
          "canQueueRun": false,
          "canReadSettings": true,
          "canReadStateVersions": true,
          "canReadVariable": true,
          "canUnlock": false,
          "canUpdate": true,
          "canUpdateVariable": true,
        },
        "planDurationAverage": 28000,
        "queueAllRuns": false,
        "resourceCount": 75,
        "runFailures": 21,
        "source": "tfe-ui",
        "speculativeEnabled": true,
        "terraformVersion": "0.14.7",
        "triggerPrefixes": Array [],
        "updatedAt": "2021-07-16T14:03:57.005Z",
        "workingDirectory": "",
        "workspaceKpisRunsCount": 30,
      },
    },
  ],
  "_type": "tfe_workspace",
  "actionsIsDestroyable": true,
  "allowDestroyPlan": true,
  "applyDurationAverage": 34000,
  "autoApply": false,
  "autoDestroyAt": undefined,
  "createdAt": "2021-02-17T21:26:58.799Z",
  "createdOn": 1613597218799,
  "description": "",
  "displayName": "integration-development-google-cloud",
  "environment": "default",
  "executionMode": "remote",
  "fileTriggersEnabled": false,
  "globalRemoteState": true,
  "latestChangeAt": "2021-05-28T17:39:48.509Z",
  "locked": false,
  "name": "integration-development-google-cloud",
  "operations": true,
  "permissions.canCreateStateVersions": false,
  "permissions.canDestroy": true,
  "permissions.canForceUnlock": false,
  "permissions.canLock": false,
  "permissions.canManageTags": true,
  "permissions.canQueueApply": false,
  "permissions.canQueueDestroy": false,
  "permissions.canQueueRun": false,
  "permissions.canReadSettings": true,
  "permissions.canReadStateVersions": true,
  "permissions.canReadVariable": true,
  "permissions.canUnlock": false,
  "permissions.canUpdate": true,
  "permissions.canUpdateVariable": true,
  "planDurationAverage": 28000,
  "policyCheckFailures": undefined,
  "queueAllRuns": false,
  "resourceCount": 75,
  "runFailures": 21,
  "source": "tfe-ui",
  "sourceName": undefined,
  "sourceUrl": undefined,
  "speculativeEnabled": true,
  "structuredRunOutputEnabled": undefined,
  "terraformVersion": "0.14.7",
  "triggerPrefixes": Array [],
  "updatedAt": "2021-07-16T14:03:57.005Z",
  "vcsRepoBranch": undefined,
  "vcsRepoDisplayIndentifier": undefined,
  "vcsRepoIdentifier": undefined,
  "vcsRepoIngressSubmodules": undefined,
  "vcsRepoOauthTokenId": undefined,
  "vcsRepoRepositoryHttpUrl": undefined,
  "vcsRepoServiceProvider": undefined,
  "vcsRepoWebhookUrl": undefined,
  "webLink": "https://app.terraform.io/app/integration-development-google-cloud/workspaces",
  "workingDirectory": "",
  "workspaceKpisRunsCount": 30,
}
`;
